### PR TITLE
Jetpack connect: Refactor jetpack-connect-notices

### PR DESCRIPTION
**Pull request by sirreal**
This was a required PR before moving files for #12991. However, lint issues have been solved by #13599 (thanks @tyxla).

I've taken this opportunity to apply a few janitorial changes, but I'd be just as happy to close this PR and move on.

To test:

* Use this branch.
* Go to `/jetpack/connect`
* Enter your site's URL.
* `/jetpack/connect/url=[ mysiteurl ]` is a handy alternative
* Ensure there are no regressions. Watch the console (preserve log) for any errors/warnings.
* Connect sites under different conditions:
  * wordpress.com site: http://calypso.localhost:3000/jetpack/connect/?url=$site
  * Connected but for different user http://calypso.localhost:3000/jetpack/connect/?url=$site
  * wordpress.com http://calypso.localhost:3000/jetpack/connect/?url=wordpress.com
  * not jetpack http://calypso.localhost:3000/jetpack/connect/?url=google.com
  * invalid http://calypso.localhost:3000/jetpack/connect/?url=alsdkjfalksdjf
  * Outdated Jetpack (`<3.9.6`)
  * Jetpack not installed
  * Jetpack not activated
  * Many more!

---

### Outdated

Update to es6 class.
Fix usage of this.translate.
Fix/improve propTypes
Remaps bad noticeTypes:
* `jetpackIsDisconnected` -> `notConnectedJetpack`
* `jetpackIsValid ` -> `alreadyOwned`
Removes some irrelevant code.
Fix lint errors must be fixed before jetpack-connect can be moved (#12991).

To test:

* Use this branch.
* Go to `/jetpack/connect`
* Enter your site's URL.
* `/jetpack/connect/url=[ mysiteurl ]` is a handy alternative
* Ensure there are no regressions. Watch the console (preserve log) for any errors/warnings.
* Connect sites under different conditions:
  * wordpress.com site: http://calypso.localhost:3000/jetpack/connect/url=jonsurrell.wordpress.com
  * Connected but for different user http://calypso.localhost:3000/jetpack/connect/url=jonsurrell.wpsandbox.me
  * wordpress.com http://calypso.localhost:3000/jetpack/connect/url=wordpress.com
  * not jetpack http://calypso.localhost:3000/jetpack/connect/url=google.com
  * invalid http://calypso.localhost:3000/jetpack/connect/url=alsdkjfalksdjf
  * Outdated Jetpack (`<3.9.6`)
  * Jetpack not installed
  * Jetpack not activated
  * Many more!

This got much bigger than I expected, probably needs to be broken down.

### CODE REVIEW

**Comment by sirreal** [comment]
I'm going to postpone this until #13599 is finished.

**Comment by sirreal** [comment]
@seear Changed to `switch` in 82a920a

**Comment by seear** [comment]
> Changed to switch in 82a920a

I was more thinking an object:

```
const notices = {
   notWordPress: {
      icon: 'block',
      text: translate( 'That\'s not a WordPress site.' ),
   },
   notActiveJetpack: // and so on...
```

Then `getNoticeValues()` becomes a lookup in the object and the propTypes for `noticeType` can be `Object.keys( notices )`.

This is just a personal preference, really. `if` or `switch` is fine.

**Comment by sirreal** [comment]
> Then getNoticeValues() becomes a lookup in the object and the propTypes for noticeType can be Object.keys( notices ).

I obviously didn't understand this after your first comment. Removing the explicit string duplication and having "automatic" `propTypes` is quite nice.

That said, I see some drawbacks:

* We need to access `this.props` for `translate` and `url` within those notices. There's no obvious way to do that on a static property or in a `const` object outside the class. That means we need to create a new object that's essentially the same for every instance. I don't see any very appealing workarounds for this. I believe this actually negates the possibility of using this object's keys to remove propTypes string duplication.
* We lose the easy and obvious cascades in switch cases, only relevant for `alreadyOwned` and `alreadyConnected`. We end up duplicating these objects.
* Originally we created an object, mutated in depending on some props, then maybe mutated it again depending on some other props. I believe I captured this behavior in a cleaner way which can be maintained regardless of whether the switch/object lookup is maintained.

I did make these changes to see what the result is.

@seear Please have a look and see if 21329972a is in line with what you had in mind and whether you think it's an improvement. I may be missing some ways it can be improved, but right now I think the drawbacks I mentioned are significant and I'd opt for the previous switch implementation.

File: client/signup/jetpack-connect/jetpack-connect-notices.jsx [hunk 1]
```diff
@@ -13,120 +13,161 @@ import { urlToSlug } from 'lib/url';
...
+		},
+
+		'wordpress.com': {
+			text: this.props.translate( 'Oops, that\'s us.' ),
+			status: 'is-warning',
+			icon: 'status',
+		},
+
+		retryingAuth: {
+			text: this.props.translate( 'Error authorizing. Page is refreshing for another attempt.' ),
```
Comment (a8ci18n) [inline client/signup/jetpack-connect/jetpack-connect-notices.jsx]: Hi! I've found a possible matching string that has already been translated **17** times:
`translate( 'Error authorizing. Page is refreshing for an other attempt.' )` *ES Score: 15*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

**Comment by seear** [comment]
> but right now I think the drawbacks I mentioned are significant and I'd opt for the previous switch implementation.

Agree 👍

**Pull request merged**